src/dispatchers/jsonPipeDispatcher.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as channels from '../protocol/channels';
+import { Dispatcher, DispatcherScope } from './dispatcher';
+import { createGuid } from '../utils/utils';
+import { serializeError } from '../protocol/serializers';
+
+export class JsonPipeDispatcher extends Dispatcher<{ guid: string }, channels.JsonPipeInitializer, channels.JsonPipeEvents> implements channels.JsonPipeChannel {
+  constructor(scope: DispatcherScope) {
+    super(scope, { guid: 'jsonPipe@' + createGuid() }, 'JsonPipe', {});
+  }
+
+  async send(params: channels.JsonPipeSendParams): Promise<channels.JsonPipeSendResult> {
+    this.emit('message', params.message);
+  }
+
+  async close(): Promise<void> {
+    this.emit('close');
+    this._dispatchEvent('closed', {});
```

Comment:
We need `if (!this._disposed)` here as well, just in case `close` command comes shortly after the pipe has disposed. Perhaps the same with `send`?